sharc/simulation.py

```diff
@@ -590,26 +642,26 @@ def calculate_bw_weights(self, bw_imt: float, bw_sys: float, ue_k: int) -> np.ar
         -------
             K-dimentional array of weights
         """
-
-        if bw_imt <= bw_sys:
-            weights = np.ones(ue_k)
-
-        elif bw_imt > bw_sys:
-            weights = np.zeros(ue_k)
-
-            bw_per_rbg = bw_imt / ue_k
-
-            # number of resource block groups that will have weight equal to 1
-            rb_ones = math.floor(bw_sys / bw_per_rbg)
-
-            # weight of the rbg that will generate partial interference
-            rb_partial = np.mod(bw_sys, bw_per_rbg) / bw_per_rbg
-
-            # assign value to weight array
-            weights[:rb_ones] = 1
-            weights[rb_ones] = rb_partial
-
-        return weights
+        ue_min_f = fc_ue - bw_ue / 2
+        ue_max_f = fc_ue + bw_ue / 2
+
+        sys_min_f = fc_sys - bw_sys / 2
+        sys_max_f = fc_sys + bw_sys / 2
+
+        # print("ue_min_f", ue_min_f)
+        # print("ue_max_f", ue_max_f)
+        # print("sys_min_f", sys_min_f)
+        # print("sys_max_f", sys_max_f)
+
+        overlap = np.maximum(
+            0,
+            np.minimum(ue_max_f, sys_max_f) - np.maximum(ue_min_f, sys_min_f)
+        ) / bw_ue
+        # print("fc_ue", fc_ue)
+        # print("overlap", overlap)
+        # exit()
+
+        return overlap
```

Comment:
Tirar os comentários

```suggestion
        ue_min_f = fc_ue - bw_ue / 2
        ue_max_f = fc_ue + bw_ue / 2

        sys_min_f = fc_sys - bw_sys / 2
        sys_max_f = fc_sys + bw_sys / 2

        overlap = np.maximum(
            0,
            np.minimum(ue_max_f, sys_max_f) - np.maximum(ue_min_f, sys_min_f)
        ) / bw_ue

        return overlap
```